bundles/org.openhab.binding.solarman/src/main/java/org/openhab/binding/solarman/internal/modbus/SolarmanV5Protocol.java

```diff
@@ -0,0 +1,270 @@
+                logger.error("Modbus frame is too short or empty");
+            }
+
+            return registers;
+        }
+
+        int actualCrc = ByteBuffer.wrap(frame, expectedFrameDataLen, 2).order(ByteOrder.LITTLE_ENDIAN).getShort()
+                & 0xFFFF;
+        int expectedCrc = CRC16Modbus.calculate(Arrays.copyOfRange(frame, 0, expectedFrameDataLen));
+
+        if (actualCrc != expectedCrc) {
+            if (allowLogging) {
+                logger.error("Modbus frame crc is not valid. Expected {}, got {}", String.format("%04x", expectedCrc),
+                        String.format("%04x", actualCrc));
+            }
+
+            return registers;
+        }
+
+        for (int i = 0; i < regCount; i++) {
+            int p1 = 3 + (i * 2);
+            ByteBuffer order = ByteBuffer.wrap(frame, p1, 2).order(ByteOrder.BIG_ENDIAN);
+            byte[] array = new byte[] { order.get(), order.get() };
+            registers.put(i + firstReg, array);
+        }
+
+        return registers;
+    }
+
+    protected byte @Nullable [] extractModbusResponseFrame(byte @Nullable [] responseFrame, byte[] requestFrame,
+            Boolean allowLogging) {
+        if (responseFrame == null || responseFrame.length == 0) {
+            if (allowLogging) {
+                logger.error("No response frame");
+            }
+
+            return null;
+        } else if (responseFrame.length == 29) {
+            parseResponseErrorCode(responseFrame, requestFrame);
+            return null;
+        } else if (responseFrame.length < (29 + 4)) {
+            if (allowLogging) {
+                logger.error("Response frame is too short");
+            }
+            return null;
+        } else if (responseFrame[0] != (byte) 0xA5) {
+            if (allowLogging) {
+                logger.error("Response frame has invalid starting byte");
+            }
+            return null;
+        } else if (responseFrame[responseFrame.length - 1] != (byte) 0x15) {
+            if (allowLogging) {
+                logger.error("Response frame has invalid ending byte");
+            }
+            return null;
+        }
+
+        return Arrays.copyOfRange(responseFrame, 25, responseFrame.length - 2);
+    }
+
+    protected void parseResponseErrorCode(byte[] responseFrame, byte[] requestFrame) {
+        if (responseFrame[0] == (byte) 0xA5 && responseFrame[1] == (byte) 0x10
+                && !Arrays.equals(Arrays.copyOfRange(responseFrame, 7, 11), Arrays.copyOfRange(requestFrame, 7, 11))) {
+            String requestInverterId = parseInverterId(requestFrame);
+            String responseInverterId = parseInverterId(responseFrame);
+
+            logger.error("There was a mismatch between the request logger ID: {} and the response logger ID: {}. "
+                    + "Make sure you are using the logger ID and not the inverter ID. "
+                    + "If in doubt, try the one in the response", requestInverterId, responseInverterId);
```

Comment:
```suggestion
            logger.debug("There was a mismatch between the request logger ID: {} and the response logger ID: {}. "
                    , requestInverterId, responseInverterId);
```

If you want to inform the user, maybe throwing a custom exception `SolarmanConfigurationException` is better as this can have a detailed message included and can be routed to a ThingStatusDetail on caller / Handler level